src/components/views/location/LocationViewDialog.tsx

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import { MatrixEvent } from 'matrix-js-sdk/src/models/event';
+
+import { replaceableComponent } from "../../../utils/replaceableComponent";
+import BaseDialog from "../dialogs/BaseDialog";
+import { IDialogProps } from "../dialogs/IDialogProps";
+import { createMap, LocationBodyContent, locationEventGeoUri, parseGeoUri } from '../messages/MLocationBody';
+
+interface IProps extends IDialogProps {
+    mxEvent: MatrixEvent;
+}
+
+interface IState {
+    error: Error;
+}
+
+@replaceableComponent("views.location.LocationViewDialog")
+export default class LocationViewDialog extends React.Component<IProps, IState> {
+    private coords: GeolocationCoordinates;
+
+    constructor(props: IProps) {
+        super(props);
+
+        this.coords = parseGeoUri(locationEventGeoUri(this.props.mxEvent));
+        this.state = {
+            error: undefined,
+        };
+    }
+
+    componentDidMount() {
+        if (this.state.error) {
+            return;
+        }
+
+        createMap(
+            this.coords,
+            true,
+            this.getBodyId(),
+            this.getMarkerId(),
+            (e: Error) => this.setState({ error: e }),
+        );
+
+        // Set class mx_Dialog_nonDialogButton on all buttons inside
+        // the map container. This prevents our CSS from styling the
+        // attribution button as if it were a dialog submit/cancel button.
+        const container: Element = document.getElementById(this.getBodyId());
+        container.querySelectorAll("button").forEach(
+            (b: Element) => b.classList.add("mx_Dialog_nonDialogButton"),
+        );
```

Comment:
this isn't great, can we not change the CSS to not apply that rule if there's a certain class on the parent/dialog?

Reply:
The CSS in question is 9 sections of [_common.scss](https://github.com/matrix-org/matrix-react-sdk/blob/develop/res/css/_common.scss#L426), that look like this kind of thing:

```css
.mx_Dialog button:not(.mx_Dialog_nonDialogButton),
.mx_Dialog input[type="submit"],
.mx_Dialog_buttons button:not(.mx_Dialog_nonDialogButton),
.mx_Dialog_buttons input[type="submit"] {
```

So it looked like I would need to add another `not()` clause to each of those 9 places, and this felt like a better solution - adding the class that expresses what I want to the element.  It's nasty because I don't directly control the element, so I have to patch it in later.

If you can see a better way, or you think adding 9 `not()` clauses is better, let me know.

Reply:
Well the above is very much a thing which is frowned upon in React so best to be avoided even if that means updating 9 selectors.

Reply:
Turned out only 5 were needed.  Done in a5991af0a9 - thanks!